README.md

```diff
@@ -245,7 +245,25 @@ To use your own match hooks, instantiate the replace matcher with a module conta
     span._.suggestions
     # >>> ['acccepts']
 ```
+#### Context matching
 
+Currently replaCy only supports string-based context using match hooks. The way to do this is by using the match hooks part_of_phrase and sentence_has. The match hook part_of_phrase checks if the matched phrase is part of the given input phrase to the hook, and sentence_has checks whether or not the sentence including the matched phrase contains the word or words given as input. For example, if you wanted to replace the word "make" with the word "brew" when appropriate, such as "The coffee was made" -> "the coffee was brewed" you could use the following match hooks. This wouldn't be perfect for all applications, such as when the sentence contains multiple instances of the word "made", such as "he made coffee and also made cookies", but works for most applications.
+
+```        
```

Comment:
IIUC, the JSON snippet shows the match hooks for a match where we want to replace `make` with `brew` when `tea` `coffee` or `beer` are in the sentence, and unless `make` is part of the phrase `make coffee`? I realize this was my example, but I want to make sure the docs are as clear as possible, since `replaCy` has more functionality than most OSS libs I have put out there, and I think it is easy to confuse the user.

the way I tested this PR was by adding the following entry to the `match_dict.json` and seeing if it worked as expected. Maybe it is a more clear example? I prefer to show an entire match in the docs:

```json
{
    ...,
    "apples-example": {
        "patterns": [
            {
                "LEMMA": "apple",
                "TEMPLATE_ID": 1
            }
        ],
        "suggestions": [
            [
                {
                    "TEXT": "malus"
                },
                {
                    "TEXT": "domestica",
                    "FROM_TEMPLATE_ID": 1
                }
            ]
        ],
        "match_hook": [
            {
                "name": "part_of_phrase",
                "args": "hungry for apples",
                "match_if_predicate_is": false
            },
            {
                "name": "sentence_has",
                "args": [
                    "rick",
                    "morty",
                    "jerry",
                    "wubba lubba"
                ],
                "match_if_predicate_is": false
            }
        ],
        "comment": "Change apple to its scientific name unless part of a Rick and Morty reference"
    }
}
```